src/bsc/BscProjectThreaded.ts

```diff
@@ -26,12 +27,38 @@ export class BscProjectThreaded implements ExtractMethods<BscProject> {
         return this.activateDeferred.isCompleted;
     }
 
+    /**
+     * If an error occurs at any time during the worker's lifetime, it will be caught and stored here.
+     */
+    public isErrored() {
+        return this.errorDeferred.isCompleted;
+    }
+
+    private ready() {
+        return Promise.race([
+            //if we've encountered an error, reject immediately. The underlying error should propagate up
+            this.errorDeferred.promise,
+            //wait for the activate to finish. This should typically be the only promise that resolves
+            this.activateDeferred.promise
+        ]);
+    }
+
     public async activate(options: Parameters<ProgramBuilder['run']>[0]) {
         const timeEnd = this.logger.timeStart('log', 'activate');
 
         // start a new worker thread or get an unused existing thread
         this.worker = bscProjectWorkerPool.getWorker();
 
+        //!!!IMPORTANT!!! this observer must be registered in order to prevent the worker thread from crashing the main thread
+        this.worker.on('error', (error) => {
```

Comment:
This was the most important part of the fix. by not observing this, node sees an unhandled exception and crashes.